code/modules/xenoarchaeology/traits/xenoartifact_activators.dm

```diff
@@ -0,0 +1,151 @@
+///Activation traits - only used to generate charge
+///============
+/// Default acvitavor, on-use / interact
+///============
+/datum/xenoartifact_trait/activator/impact
+	desc = "Sturdy"
+	label_desc = "Sturdy: The material is sturdy. The amount of force applied seems to directly correlate to the size of the reaction."
+	charge = 25
+	signals = list(COMSIG_PARENT_ATTACKBY, COMSIG_MOVABLE_IMPACT, COMSIG_ITEM_ATTACK_SELF, COMSIG_ITEM_AFTERATTACK)
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+	weight = 30
+
+/datum/xenoartifact_trait/activator/impact/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target)
+	var/obj/item/xenoartifact/X = source
+	charge = charge*((thing?.force || 10)*0.1)
+	X.default_activate(charge, user, target)
+
+///============
+/// Burn activator, responds to heat
+///============
+/datum/xenoartifact_trait/activator/burn
+	desc = "Flammable"
+	label_desc = "Flammable: The material is flammable, and seems to react when ignited."
+	charge = 25
+	signals = list(COMSIG_PARENT_ATTACKBY)
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/activator/burn/on_init(obj/item/xenoartifact/X)
+	..()
+	X.max_range += 1
+
+/datum/xenoartifact_trait/activator/burn/calculate_charge(datum/source, obj/item/thing, mob/user, atom/target, params) //xenoa item handles this, see process proc there
+	var/obj/item/xenoartifact/X = source
+	if(X.process_type != PROCESS_TYPE_LIT && thing.ignition_effect(X, user))
+		X.visible_message("<span class='danger'>The [X.name] sparks on.</span>")
+		X.process_type = PROCESS_TYPE_LIT
+		sleep(1.8 SECONDS) //Give them a chance to escape
+		START_PROCESSING(SSobj, X)
+		log_game("[key_name_admin(user)] lit [X] at [world.time] using [thing]. [X] located at [X.x] [X.y] [X.z].")
+
+///============
+/// Timed activator, activates on a timer. Timer is turned on when used, has a chance to turn off.
+///============
+/datum/xenoartifact_trait/activator/clock
+	label_name = "Tuned"
+	label_desc = "Tuned: The material produces a resonance pattern similar to quartz, causing it to produce a reaction every so often."
```

Comment:
This name is very confusing and makes it hard to find. My main issue with a lot of these are the confusing names

Reply:
> Timed